slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -63,25 +63,27 @@ trait PostgresProfile extends JdbcProfile {
       override def schema = super.schema.filter(_ != "public") // remove default schema
     }
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
-      val VarCharPattern = "^'(.*)'::character varying$".r
-      val TextPattern = "^'(.*)'::text".r
-      val IntPattern = "^\\((-?[0-9]*)\\)$".r
+      val NumericPattern = "^['(]?(-?[0-9]+\\.?[0-9]*)[')]?(?:::(?:numeric|bigint|integer))?".r
+      val TextPattern = "^'(.*)'::(?:bpchar|character varying|text)".r
+      val UUIDPattern = "^'(.*)'::uuid".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case ("true","Boolean")  => Some(Some(true))
         case ("false","Boolean") => Some(Some(false))
-        case (VarCharPattern(str),"String") => Some(Some(str))
         case (TextPattern(str),"String") => Some(Some(str))
-        case (IntPattern(v),"Int") => Some(Some(v.toInt))
-        case (IntPattern(v),"Long") => Some(Some(v.toLong))
-        case ("NULL::character varying","String") => Some(None)
```

Comment:
https://github.com/AtkinsChang/slick/blob/e53c400c0edf4fb24f55009564e2885ecb7e2712/slick/src/main/scala/slick/jdbc/PostgresProfile.scala#L97
I remove the match to `NULL::character varying` because it can't reproduce in postgres 9.X. AFAIK `NULL::character varying` appears before postgres 8.X which is 6 years ago. I can add the match back if you think we should support to the very old version.